runner/check.go

```diff
@@ -44,9 +45,18 @@ func (c *Check) Run(ctx context.Context, role string, env []string) ([]byte, int
 	newCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 
-	cmd := exec.CommandContext(newCtx, c.Cmd...)
-	cmd.Env = env
-	stdout, stderr, code, err := exec.FullOutput(cmd)
+	// Set up check environment
+	for varName, newVal := range envOverride {
+		oldVal, set := os.LookupEnv(varName)
+		if set {
+			defer os.Setenv(varName, oldVal)
+		} else {
+			defer os.Unsetenv(varName)
+		}
+		os.Setenv(varName, newVal)
+	}
+
+	stdout, stderr, code, err := exec.FullOutput(exec.CommandContext(newCtx, c.Cmd...))
```

Comment:
I liked the version better where you were simply assigning `cmd.Env` to env.  That way you don't pollute the env for the entire process.  Would it be better to basically do something like this?

```
newEnv := []string
for _, e := range os.Environ() {
  newEnv = append(newEnv,e)
}
for k,v := range envOverride {
  newEnv = append(newEnv, k + "=" + v)
}
cmd := exec.CommandContext(newCtx, c.Cmd...)
cmd.Env = newEnv
....
```

Reply:
Me too, unfortunately I really do need to modify the parent process's env before starting the subprocess -- it's the parent process's `PATH` that gets used when searching for the child process's executable. If I went with the approach you suggest, the unit test I updated in this PR would fail because it can't find `combined.sh`.

I could have the runner config distinguish between parent and child process environments, but I couldn't come up with a good use case for it, so it seemed like unnecessary complexity.

Reply:
I changed this so that now the environment gets set when the runner process starts. This should eliminate any possible issues around concurrency.